File: ios/Calculator.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(Calculator, NSObject)

RCT_EXTERN_METHOD(add:(double)a
                  b:(double)b
                  resolve:(RCTPromiseResolveBlock)resolve
                  reject:(RCTPromiseRejectBlock)reject)

@end

File: ios/Calculator.swift
import Foundation
import React

@objc(Calculator)
class Calculator: NSObject {

  @objc
  func add(_ a: Double,
           b: Double,
           resolve: RCTPromiseResolveBlock,
           reject: RCTPromiseRejectBlock) {
    let result = a + b
    resolve(result)
  }

  @objc
  static func requiresMainQueueSetup() -> Bool {
    return false
  }
}